src/parser/GenericParser.test.ts

```diff
@@ -0,0 +1,281 @@
+                    language: Language.CPlusPlus,
+                    tree: tree,
+                });
+            });
+
+        const calculateMetricsSpied = jest
+            .spyOn(MetricCalculator.prototype, "calculateMetrics")
+            .mockResolvedValue(["clearly/invalid/path.cpp", expectedFileMetricsMap]);
+
+        const parser = new GenericParser(getParserConfiguration("clearly/invalid/path.cpp"));
+        const actualResult = await parser.calculateMetrics();
+
+        expect(actualResult.fileMetrics).toEqual(
+            new Map([["clearly/invalid/path.cpp", expectedFileMetricsMap]])
+        );
+        expect(calculateMetricsSpied).toHaveBeenCalledTimes(1);
+        expect(treeParserSpied).toHaveBeenCalledTimes(1);
+    });
+
+    it("should return the appropriate values and call the appropriate functions in case there is no error and two files are found", async () => {
+        const expectedFileMetricsMap = new Map([
+            [FileMetric.linesOfCode, { metricName: FileMetric.linesOfCode, metricValue: 5 }],
+        ]);
+
+        const findFilesAsyncMock = findFilesAsync as jest.Mocked<typeof findFilesAsync>;
+        findFilesAsyncMock.mockImplementation((config) => {
+            return mockedFindTwoFiles();
+        });
+
+        const treeParserSpied = jest
+            .spyOn(TreeParser, "parse")
+            .mockImplementation((file, assumedLanguage) => {
+                return Promise.resolve({
+                    fileExtension: file.fileExtension,
+                    filePath: file.filePath,
+                    language: Language.CPlusPlus,
+                    tree: tree,
+                });
+            });
+
+        const calculateMetricsSpied = jest
+            .spyOn(MetricCalculator.prototype, "calculateMetrics")
+            .mockImplementation(async (parsedFilePromise) => [
+                (await parsedFilePromise).filePath,
+                expectedFileMetricsMap,
+            ]);
+
+        const parser = new GenericParser(getParserConfiguration("clearly/invalid"));
+        const actualResult = await parser.calculateMetrics();
+
+        expect(actualResult.fileMetrics).toEqual(
+            new Map([
+                ["clearly/invalid/path1.cc", expectedFileMetricsMap],
+                ["clearly/invalid/path2.cpp", expectedFileMetricsMap],
+            ])
+        );
+        expect(calculateMetricsSpied).toHaveBeenCalledTimes(2);
+        expect(treeParserSpied).toHaveBeenCalledTimes(2);
+    });
+
+    it("should return the appropriate value and call the appropriate functions in case there is an error while parsing the tree", async () => {
+        const expectedFileMetricsMap = new Map([
+            ["ERROR", { metricName: "ERROR", metricValue: -1 }],
+        ]);
+
+        const findFilesAsyncMock = findFilesAsync as jest.Mocked<typeof findFilesAsync>;
+        findFilesAsyncMock.mockImplementation((config) => {
+            return mockedFindFilesAsync();
+        });
```

Comment:
Could be done in a beforeEach I guess
```suggestion
        mockFindFilesAsync()
```